spec/SensioLabs/Behat/PageObjectExtension/PageObject/Factory/CamelcasedClassNameResolverSpec.php

```diff
@@ -77,4 +86,22 @@ function it_throws_an_exception_if_element_class_does_not_exist()
         $this->shouldThrow(new \InvalidArgumentException($expectedMessage))
             ->duringResolveElement('Recent searches');
     }
+
+    function it_resolves_an_element_class_given_as_fqcn()
+    {
+        $this->beConstructedWith(array('\\Foo', '\\Element'));
+
+        $elementFQCN = 'spec\\SensioLabs\\Behat\\PageObjectExtension\\PageObject\\Factory\\Bar';
+        $this->resolvePage($elementFQCN)->shouldReturn($elementFQCN);
+    }
+}
+
+class Foo extends Page
```

Comment:
There's still a difference, as in the existing examples only part of the namespace is passed (and a whole namespace is returned with prefixes). In your new examples, you'd be passing a full namespace in, and expecting it back.